neuraxle/metaopt/random.py

```diff
@@ -25,27 +25,264 @@
+
+        self._update_scores_validation(validation_data_inputs, validation_expected_outputs)
+
+        return self
+
+    def _update_scores_validation(self, data_inputs, expected_outputs):
+        self.scores_validation = self.scoring_function(expected_outputs, data_inputs)
+        self.scores_validation_mean = np.mean(self.scores_validation)
+        self.scores_validation_std = np.std(self.scores_validation)
+
+    def _update_scores_train(self, data_inputs, expected_outputs):
+        self.scores_train = self.scoring_function(expected_outputs, data_inputs)
+        self.scores_train_mean = np.mean(self.scores_train)
+        self.scores_train_std = np.std(self.scores_train)
+
+    def split_data_container(self, data_container) -> Tuple[DataContainer, DataContainer]:
+        """
+        Split data container into a training set, and a validation set.
+
+        :param data_container: data container
+        :type data_container: DataContainer
+        :return: train_data_container, validation_data_container
+        """
+
+        train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs = \
+            self.split(data_container.data_inputs, data_container.expected_outputs)
+
+        train_ids = self.train_split(data_container.current_ids)
+        train_data_container = DataContainer(train_ids, train_data_inputs, train_expected_outputs)
+
+        validation_ids = self.validation_split(data_container.current_ids)
+        validation_data_container = DataContainer(validation_ids, validation_data_inputs, validation_expected_outputs)
+
+        return train_data_container, validation_data_container
+
+    def split(self, data_inputs, expected_outputs=None) -> Tuple[List, List, List, List]:
+        """
+        Split data inputs, and expected outputs into a training set, and a validation set.
+
+        :param data_inputs: data inputs to split
+        :param expected_outputs: expected outputs to split
+        :return: train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs
+        """
+        validation_data_inputs = self.validation_split(data_inputs)
+        validation_expected_outputs = None
+        if expected_outputs is not None:
+            validation_expected_outputs = self.validation_split(expected_outputs)
+
+        train_data_inputs = self.train_split(data_inputs)
+        train_expected_outputs = None
+        if expected_outputs is not None:
+            train_expected_outputs = self.train_split(expected_outputs)
+
+        return train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs
+
+    def train_split(self, data_inputs) -> List:
+        """
+        Split training set.
+
+        :param data_inputs: data inputs to split
+        :return: train_data_inputs
+        """
+        index_split = math.floor(len(data_inputs) * (1 - self.test_size))
+        return data_inputs[0:index_split]
```

Comment:
`index_split = math.floor(len(data_inputs) * (1 - self.test_size))` should be extracted to a method as it is duplicate code. 

Should we have a method like `BaseDataContainer.train_test_split()` which returns two data containers instead? (a train and a test one as return value). To me it feels like those two methods (def train split and def test split) should be just one and do the thing only at once to reduce risks of error and couple the good things together).